### PR TITLE
added ability to assign moderator role via admin/super admin

### LINKED ISSUE #17458: Make Moderator role assignable to user
An admin or super admin can assign the `moderator` role.

- create moderator role in database
- modify `handle_user_status` method to assign role when selected
- update specs accordingly

### PR DESCRIPTION
**Pull request by dscottS3**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This PR allows a super admin to assign the moderator role to a user. NOTE: this does NOT enforce the moderator role, yet.

## Related Tickets & Documents

- Closes #17458 

## QA Instructions, Screenshots, Recordings

Enable the `moderator_role` feature flag and verify that the role is selectable in the roles drop down (special roles).
Assign a user the moderator role and ensure the moderator role is displayed when managing/editing that user.
_Note: When running Cypress tests, choose 'y' to rebuild the test database, so the test-moderator role is created._

### UI accessibility concerns?

no -- only a backend change

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

File: app/models/role.rb [hunk 1]
```diff
@@ -15,7 +17,6 @@ class Role < ApplicationRecord
     trusted
     warned
     workshop_pass
-    creator
```
Comment (dscottS3) [inline app/models/role.rb:18]: alphabetizing this list

**Review by Zhao-Andy — APPROVED** [review]
LGTM!

**Review by fdocr — APPROVED** [review]
The PR looks awesome! LGTM

**non-blocking question**: Should the [`BASE_ROLES` constant](https://github.com/forem/forem/blob/dscottS3/make-mod-role-assignable-to-user-17458/app/lib/constants/role.rb#L3) include the moderator role? I'm not sure if that is part of the scope of this PR or will that be handled separately (just noticed the related issues' mention of a Feature Flag).

Asking because I attempted to assign the role via the admin dashboard and couldn't find it in the dropdown there, which I believe is populated by that constant.

Screen Shot 2022-05-25 at 15 44 45

**Comment by fdocr** [comment]
Also just noticed Travis seems stuck here. I had to close and re-open my PR (#17761) for Travis to pick it back up earlier today, in case that may help 😄

**Comment by msarit** [comment]
> non-blocking question: Should the [BASE_ROLES constant](https://github.com/forem/forem/blob/dscottS3/make-mod-role-assignable-to-user-17458/app/lib/constants/role.rb#L3) include the moderator role? I'm not sure if that is part of the scope of this PR or will that be handled separately (just noticed the related issues' mention of a Feature Flag).

@fdocr yes the feature flag needs to be enabled to see the Moderator role (under Special Roles) in /admin

File: spec/support/seeds/seeds_e2e.rb [hunk 2]
```diff
@@ -71,6 +71,33 @@
 
 ##############################################################################
 
+seeder.create_if_doesnt_exist(User, "email", "moderator-user@forem.local") do
```
Comment (aitchiss) [inline spec/support/seeds/seeds_e2e.rb:74]: I think you are running into some similar issues as @msarit did with the E2E test failures - there's a spec that assumes the 'trusted-user-1@forem.local' below is user with `id: 2`, but after adding this one they're `id: 3` 🙃 

I mentioned to Arit that probably that test wants reworked so it doesn't rely on the ID, but I think for a quick fix if you create this user _after_ the one immediately below you should see some of those specs just magically start passing again

Comment (msarit) [reply spec/support/seeds/seeds_e2e.rb:74]: Thank you Suzanne! 🙏🏾

**Pull request merged**